src/main/java/com/gojek/beast/protomapping/ProtoUpdateListener.java

```diff
@@ -55,24 +87,39 @@ public void onProtoUpdate() {
         log.info("updating bq table as {} proto was updated", getProto());
         try {
             updateProtoParser();
-        } catch (ExternalCallException e) {
-            log.warn("Error while updating bigquery table:" + e.getMessage());
+        } catch (ProtoNotFoundException | BigquerySchemaMappingException | BigQueryException e) {
+            String errMsg = "Error while updating bigquery table:" + e.getMessage();
+            log.error(errMsg);
+            e.printStackTrace();
+            throw new BQTableUpdateFailure(errMsg);
         }
     }
 
     // First get latest protomapping, update bq schema, and if all goes fine
     // then only update beast's proto mapping config
-    private void updateProtoParser() throws ExternalCallException {
-        UpdateBQTableRequest request = new UpdateBQTableRequest(appConfig.getGCPProject(), proto, appConfig.getTable(), appConfig.getDataset());
-        String protoMapping = updateTableService.getProtoMappingFromRemoteURL(protoMappingConfig.getProtoColumnMappingURL(), proto);
-        updateTableService.updateBigQuerySchema(protoMappingConfig.getUpdateBQTableURL(), request);
-        protoMappingConfig.setProperty("PROTO_COLUMN_MAPPING", protoMapping);
+    private void updateProtoParser() throws ProtoNotFoundException, BigquerySchemaMappingException, BigQueryException {
+        ProtoField protoField = protoFieldFactory.getProtoField();
+        protoField = protoMappingParser.parseFields(protoField, proto, stencilClient);
+        JsonObject protoMappingJson = protoMappingConverter.generateColumnMappings(protoField.getFields());
```

Comment:
`JsonObject` reference isn't needed here. The Converter should return stringified columnmapping isolating gson stuff within converter itself.